### PR TITLE
ECL-148/initial test journey automation first commit

### PR DESCRIPTION
**Pull request by jamesvarghese701**

### CODE REVIEW

File: .bsp/sbt.json [hunk 1]
```diff
@@ -0,0 +1 @@
+{"name":"sbt","version":"1.5.8","bspVersion":"2.0.0-M5","languages":["scala"],"argv":["/Library/Java/JavaVirtualMachines/jdk-17.0.2.jdk/Contents/Home/bin/java","-Xms100m","-Xmx100m","-classpath","/Users/jamesvarghese/Library/Application Support/JetBrains/IdeaIC2022.2/plugins/Scala/launcher/sbt-launch.jar","xsbt.boot.Boot","-bsp","--sbt-launch-jar=/Users/jamesvarghese/Library/Application%20Support/JetBrains/IdeaIC2022.2/plugins/Scala/launcher/sbt-launch.jar"]}
```
Comment (peter-hazell) [inline .bsp/sbt.json:1]: Please add `.bsp` to your `.gitignore` file - we shouldn't be committing this as it's specific to your machine

File: src/test/resources/features/login.feature [hunk 2]
```diff
@@ -0,0 +1,6 @@
+Feature: login to ECL
+@WIP
+  Scenario: User able to navigate to economic crime and levy page
```
Comment (peter-hazell) [inline src/test/resources/features/login.feature:3]:
```suggestion
  Scenario: User able to navigate to economic crime levy start page
```

File: src/test/scala/uk/gov/hmrc/test/ui/cucumber/stepdefs/LoginStepDef.scala [hunk 3]
```diff
@@ -14,24 +14,32 @@
...
 
-  def submitVATInformation: CheckYourVATResult.type = {
-    submitPage()
-    CheckYourVATResult
+    When("""I click on submit button""") { () =>
+
+      LoginPage.authorityWizardSubmitButton()
+    }
+
+  Then("""I am on levy screen""") { () =>
```
Comment (peter-hazell) [inline src/test/scala/uk/gov/hmrc/test/ui/cucumber/stepdefs/LoginStepDef.scala:38]:
```suggestion
  Then("""I am on the start page""") { () =>
```

File: src/test/resources/features/login.feature [hunk 4]
```diff
@@ -0,0 +1,6 @@
...
+    Given I launch the login page
+    When I click on submit button
+    Then I am on levy screen
```
Comment (peter-hazell) [inline src/test/resources/features/login.feature:6]:
```suggestion
    Then I am on the start page
```

File: src/test/scala/uk/gov/hmrc/test/ui/pages/BasePage.scala [hunk 5]
```diff
@@ -16,21 +16,45 @@
...
+  val WAIT_POLLING_INTERVAL: Duration = Duration.of(250, ChronoUnit.MILLIS)
+  val WAIT_TIME_OUT: Duration = Duration.of(20, ChronoUnit.SECONDS)
+
+  private val fluentWait: Wait[WebDriver] = new FluentWait[WebDriver](driver)
+    .withTimeout(WAIT_POLLING_INTERVAL)
+    .pollingEvery(WAIT_TIME_OUT)
+
+
+  def host(localPort: Int): String = env match {
+  //  case "qa" => "http://localhost:14000/register-for-economic-crime-levy/"
```
Comment (peter-hazell) [inline src/test/scala/uk/gov/hmrc/test/ui/pages/BasePage.scala:44]: should this be commented out?

the service is deployed to QA

File: .bsp/sbt.json (hunk 1 above)
Comment (jamesvarghese701) [reply .bsp/sbt.json]: Done

File: src/test/resources/features/login.feature (hunk 2 above)
Comment (jamesvarghese701) [reply src/test/resources/features/login.feature]: Corrections done

File: src/test/scala/uk/gov/hmrc/test/ui/cucumber/stepdefs/BaseStepDef.scala [hunk 6]
```diff
@@ -26,7 +26,7 @@ import scala.util.Try
 
 trait BaseStepDef extends ScalaDsl with EN with BrowserDriver with Eventually with Matchers {
 
-  sys.addShutdownHook {
-    Try(SingletonDriver.closeInstance)
-  }
+//  sys.addShutdownHook {
+//    Try(SingletonDriver.closeInstance)
+//  }
```
[the remark this answers is not in the document]
Comment (jamesvarghese701) [reply src/test/scala/uk/gov/hmrc/test/ui/cucumber/stepdefs/BaseStepDef.scala]: Needed that code, so removed the comment

File: src/test/scala/uk/gov/hmrc/test/ui/cucumber/stepdefs/LoginStepDef.scala (hunk 3 above)
Comment (jamesvarghese701) [reply src/test/scala/uk/gov/hmrc/test/ui/cucumber/stepdefs/LoginStepDef.scala:32]: Done

File: src/test/resources/features/login.feature (hunk 4 above)
Comment (jamesvarghese701) [reply src/test/resources/features/login.feature:7]: Done

File: src/test/scala/uk/gov/hmrc/test/ui/pages/LoginPage.scala [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+  val authoritySubmitButton = By.id("submit-top")
+
+  def navigateToUrl(): Unit = {
+
+    driver.get(authLoginPageUrl)
+  }
+
+  def authorityWizardSubmitButton(): Unit = {
+    clickElement(authoritySubmitButton)
+//    driver.findElement(authoritySubmitButton).click()
```
[the remark this answers is not in the document]
Comment (jamesvarghese701) [reply src/test/scala/uk/gov/hmrc/test/ui/pages/LoginPage.scala]: Removed

**Review by peter-hazell — APPROVED** [review]

**Pull request merged**